### ISSUE TITLE
Error convertir LocalDateTime in Java driver

### ISSUE THREAD

**Issue by bitli** [rethinkdb/rethinkdb#5228]
If you insert an object including a LocalDateTime, you get the exception:
`java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: OffsetSeconds`

This is because the method `ast.Util.toReqlAst()` uses the formatting pattern `"yyyy-MM-dd'T'HH:mm:ss.SSSX"` at line 90. The `LocatDateTime` does not support timezone at all (it is not a local timezone !).  The format should not include a timezone at all (maybe a fixed text) or the `LocalDateTime` should be converted to an `OffsetDateTime` before formatting.  

Currently the method attempts (and fails) try to insert an arbitrary timezone (the one where the program happens to execute).  This is likely confusing, as specifying a local time zone is not the same as the 'timezone is not relevant' which is the semantic of the (not very well named) LocalDateTime. But I do not known enough of the DB internals to know what is the best representation in this case.

**Comment by deontologician** [comment]
When we store a date time in the other drivers, we just convert to UTC if a timezone isn't specified. The db needs a timezone. It sounds like OffsetDateTime is what it should be converted to

**Comment by IvoLimmen** [comment]
I am testing out rethinkdb using Java and I am running:
> Running rethinkdb 2.3.5~0xenial (GCC 5.3.1)...
> Running on Linux 4.4.0-77-generic x86_64
I have the same problem with the LocalDateTime conversion. Tried to make a conversion myself by using the Date class but that resulted in:
```
com.rethinkdb.gen.exc.ReqlDriverError: Can't convert Thu Feb 09 13:30:48 CET 2017 to a ReqlAst: Make getDate of Thu Feb 09 13:30:48 CET 2017 Long instead of Integer
        at com.rethinkdb.ast.Util.toMap(Util.java:159)
        at com.rethinkdb.ast.Util.toReqlAst(Util.java:120)
        at com.rethinkdb.ast.Util.toReqlAst(Util.java:33)
        at com.rethinkdb.ast.Util.toReqlAst(Util.java:76)
        at com.rethinkdb.ast.Util.toReqlAst(Util.java:33)
        at com.rethinkdb.ast.Util.toReqlAst(Util.java:120)
        at com.rethinkdb.ast.Util.toReqlAst(Util.java:33)
        at com.rethinkdb.model.Arguments.coerceAndAdd(Arguments.java:43)
        at com.rethinkdb.gen.ast.Table.insert(Table.java:65)
        at org.limmen.mystart.RethinkDbLinks.lambda$storeCollection$0(RethinkDbLinks.java:57)
        at java.util.ArrayList.forEach(ArrayList.java:1249)
        at org.limmen.mystart.RethinkDbLinks.storeCollection(RethinkDbLinks.java:56)

```

**Comment by adriantodt** [comment]
The Java driver now lives in https://github.com/rethinkdb/rethinkdb-java, and we released v2.4.0. Looking at the code, I don't think it should happen anymore. Please open a ticket there if it there's some weird behaviour still going on.
